### PR TITLE
Try to refine the prompt to prevent split line ranges that give bad suggestions

### PR DESCRIPTION
**Pull request by zmeir**
I noticed that in some cases the `/improve` command will try to repeat the same suggestion for several code locations in the same file, for example - something like this:
```json
"Code suggestions": [
  {
    "relevant file": "pr_agent/git_providers/codecommit_client.py",
    "suggestion content": "Add type hints to the method signatures.",
    "existing code": "def get_differences(self, repo_name: int, destination_commit: str, source_commit: str):\n\ndef get_file(self, repo_name: str, file_path: str, sha_hash: str, optional: bool = False):\n\ndef get_pr(self, pr_number: int):\n\ndef publish_comment(self, repo_name: str, pr_number: int, destination_commit: str, source_commit: str, comment: str):",
    "relevant lines": "64-67, 100-103, 136-139, 167-170",
    "improved code": "def get_differences(self, repo_name: str, destination_commit: str, source_commit: str) -> List[CodeCommitDifferencesResponse]:\n\ndef get_file(self, repo_name: str, file_path: str, sha_hash: str, optional: bool = False) -> str:\n\ndef get_pr(self, pr_number: int) -> CodeCommitPullRequestResponse:\n\ndef publish_comment(self, repo_name: str, pr_number: int, destination_commit: str, source_commit: str, comment: str) -> None:"
  },
  ...
]
```

This creates some ugly and wrong suggestions in the PR. For example:  

image

I tried adding more "constraints" in the prompt to force it to split such suggestions in separate list items.

### CODE REVIEW

**Comment by mrT23** [comment]
The prompt you suggest is far too wordy.
it will come at the expense of other parts.

**Comment by zmeir** [comment]
Got it, thanks. I'll try to make it more concise.

A couple of questions:
1. Where do you think I should add it? In the "specific instructions" section or in the JSON schema description?
2. What about the example for the illegal output? Should I keep it?

**Comment by zmeir** [comment]
Oh yes, this is **much** better than my idea 😅

**Comment by zmeir** [comment]
@mrT23 I implemented your suggestion. Let me know what you think.

**Comment by mrT23** [comment]
looks good :)

**Review by mrT23 — APPROVED** [review]
